### PR TITLE
FEATURE: Implement pager, ordering and search in user management module

### PR DESCRIPTION
**Pull request by theDex**
This orders the results in the backend user management module by username and introduces a pager. Furthermore I implemented a search field to search in usernames and full names

### CODE REVIEW

**Comment by daniellienert** [comment]
@theDex: Would you please rebase your branch to current master to give the tests the chance to run through.

File: Neos.Neos/Classes/Controller/Module/Administration/UsersController.php [hunk 1]
```diff
@@ -84,13 +84,21 @@ protected function initializeAction()
     /**
      * Shows a list of all users
      *
+     * @param string $searchTerm
      * @return void
      */
-    public function indexAction()
+    public function indexAction(string $searchTerm = '')
     {
+        if (empty($searchTerm)) {
```
Comment (daniellienert) [inline Neos.Neos/Classes/Controller/Module/Administration/UsersController.php]: In the second column of the list, the username is shown, but not searchable. Would it be possible to also search by the accounts username?

File: Neos.Neos/Resources/Private/Templates/Module/Administration/Users/Index.html [hunk 2]
```diff
@@ -3,75 +3,87 @@
 
 <f:section name="content">
 	<div class="neos-row-fluid">
+		<f:form action="index">
```
Comment (daniellienert) [inline Neos.Neos/Resources/Private/Templates/Module/Administration/Users/Index.html]: Wouldn't it be nice to display some message if no user can be found?

Comment (theDex) [reply Neos.Neos/Resources/Private/Templates/Module/Administration/Users/Index.html]: Done

File: Neos.Neos/Classes/Controller/Module/Administration/UsersController.php (hunk 1 above)
Comment (theDex) [reply Neos.Neos/Classes/Controller/Module/Administration/UsersController.php]: Done

**Comment by daniellienert** [comment]
Original Message of @theDex:


**How I did it**

Two new methods in the UserRepository (for sorting and searching). Search parameter is sent via a f:form. Pagination ist done via f:paginate.

**How to verify it**

Just open the backend user management module (preferably with more than 10 Users)

**Checklist**

- [x] Code follows the PSR-2 coding style

**Pull request merged**